controllers/reconcile.go

```diff
@@ -853,6 +853,97 @@ func setupFedModules(feEnv *crd.FrontendEnvironment, frontendList *crd.FrontendL
 	return nil
 }
 
+func adjustSearchEntry(searchEntry *crd.SearchEntry, frontend crd.Frontend) crd.SearchEntry {
+	newSearchEntry := crd.SearchEntry{
```

Comment:
You might want to guard on a nil check on `searchEntry`? I know you are only calling this in a loop that is already guarded on a nil check, but if this ever ended up getting called somewhere else someday could result in a panic.

Reply:
I'll do the check in the iteration, the value must exist.